File: templates/apply.pl
#!/usr/bin/perl

# import required modules
use XML::DOM;
use XML::XSLT;

# define local variables
my $xmlfile = $ARGV[0];
my $xslfile = $ARGV[1];

# create an instance of XSL::XSLT processor
my $xslt = XML::XSLT->new($xslfile);

# transform XML file 
$xslt->transform($xmlfile);

# send to output
my $str = $xslt->toString;
print $str;

# free up some memory
$xslt->dispose(); 

File: templates/apply.py
#!/usr/bin/python
try:
  from lxml import etree
  print("running with lxml.etree")
except ImportError:
  print("import lxml.etree failed")
data = open('d1.xsl')
xslt_content = data.read()
xslt_root = etree.XML(xslt_content)
dom = etree.parse('d1.xml')
transform = etree.XSLT(xslt_root)
result = transform(dom)
print(str(result))
